MAVProxy/modules/mavproxy_console.py

```diff
@@ -514,6 +526,11 @@ def mavlink_packet(self, msg):
                 alt_error = "%d%s" % (msg.alt_error, alt_error_sign)
             self.console.set_status('AltError', 'AltError %s' % alt_error)
             self.console.set_status('AspdError', 'AspdError %.1f%s' % (msg.aspd_error*0.01, aspd_error_sign))
+        for i in range(0,len(self.user_added)):
+            if type == self.user_added[i][2]:
+                code = "self.console.set_status('"+self.user_added[i][0]+"','" +self.user_added[i][1]+"' %"+self.user_added[i][3]+", row = "+self.user_added[i][4]+")"
+                exec(code)
```

Comment:
why exec? we can just call set_status() directly after evaluating the expression
also needs to catch exceptions for bad expressions